src/main/java/seedu/address/ui/CommandBox.java

```diff
@@ -82,4 +108,80 @@ public interface CommandExecutor {
         CommandResult execute(String commandText) throws CommandException, ParseException;
     }
 
+    /**
+     * Contains all successfully executed commands from the last application start.
+     * Keeps track of the currently displayed command's index in the history.
+     */
+    class CommandHistory {
+        private ArrayList<String> commandHistory;
+        private int index;
+
+        public CommandHistory() {
+            commandHistory = new ArrayList<>();
+            commandHistory.add("");
+
+            index = 0;
+        }
+
+        /**
+         * Adds a successfully executed command into {@code commandHistory}, and sets
+         * {@code textField} to an empty string.
+         *
+         * @param command the user-inputted String of the successfully executed command.
+         */
+        public void addCommand(String command, TextField textField) {
+            addToCommandHistory(command);
+            initialiseForNextCommand();
+            textField.setText("");
+
+            index = commandHistory.size() - 1;
+        }
+
+        private void addToCommandHistory(String command) {
+            commandHistory.set(commandHistory.size() - 1, command);
+        }
+
+        private void initialiseForNextCommand() {
+            commandHistory.add("");
+        }
+
+        private boolean isMostRecentCommand() {
+            return index == commandHistory.size() - 1;
+        }
+
+        private boolean noPreviousCommand() {
+            return index <= 0;
+        }
+
+        private boolean noNextCommand() {
+            return index >= commandHistory.size() - 1;
+        }
+
+        /**
+         * Sets the text in {@code textField} to the previous command, if a previous command exists.
+         * If the current command is the most recent command, adds that to history.
+         *
+         * @param textField the TextField where the user inputs commands in.
+         */
+        public void getPreviousCommand(TextField textField) {
+            if (noPreviousCommand()) {
+                return;
+            } else if (isMostRecentCommand()) {
+                addToCommandHistory(textField.getText());
+            }
+            textField.setText(commandHistory.get(--index));
+        }
+
+        /**
+         * Sets the text in {@code textField} to the next command, if a next command exists.
+         *
+         * @param textField the TextField where the user inputs commands in.
+         */
+        public void getNextCommand(TextField textField) {
+            if (noNextCommand()) {
+                return;
+            }
```

Comment:
Link to my prev comment, this line will prob read something like
```suggestion
            if (noNextCommand()) {
                textField.setText(currentTextField)
            }
```